src/protocol/OpenConnectionRequest2.php

```diff
@@ -21,19 +21,29 @@
 class OpenConnectionRequest2 extends OfflineMessage{
 	public static $ID = MessageIdentifiers::ID_OPEN_CONNECTION_REQUEST_2;
 
+	public bool $serverSecurity = false;
+	public int $cookie = 0;
 	public int $clientID;
 	public InternetAddress $serverAddress;
 	public int $mtuSize;
 
 	protected function encodePayload(PacketSerializer $out) : void{
 		$this->writeMagic($out);
+		if ($this->serverSecurity) {
+			$out->putInt($this->cookie);
+			$out->putByte(0); // WHY MOJANG?
+		}
 		$out->putAddress($this->serverAddress);
 		$out->putShort($this->mtuSize);
 		$out->putLong($this->clientID);
 	}
 
 	protected function decodePayload(PacketSerializer $in) : void{
 		$this->readMagic($in);
+		if ($this->serverSecurity) {
+			$this->cookie = $in->getInt();
+			$in->getByte(); // JUST 5 BYTES AND THERE IS WEIRD EXTRA BYTE
```

Comment:
It's not some weird extra byte and we need to implement it.

Reply:
> It's not some weird extra byte and we need to implement it.

![Ekran görüntüsü 2024-08-03 175923](https://github.com/user-attachments/assets/92adf378-2a78-43bd-8703-06fff617d123)
I don't understand what it does.

Reply:
Documented as if client support cookie security : https://wiki.vg/Raknet_Protocol

Reply:
> Documented as if client support cookie security : https://wiki.vg/Raknet_Protocol

I never realized it was updated!